utils/utils.go

```diff
@@ -738,6 +740,21 @@ func CustomizePodSpec(pts *corev1.PodTemplateSpec, ba common.BaseComponent) {
 	pts.Spec.AutomountServiceAccountToken = &mount
 }
 
+// Initialize an empty TopologySpreadConstraints list and optionally prefers scheduling across zones for pods with zoneDomainMatchLabels
+func CustomizeTopologySpreadConstraints(pts *corev1.PodTemplateSpec, zoneDomainMatchLabels map[string]string) {
+	pts.Spec.TopologySpreadConstraints = make([]corev1.TopologySpreadConstraint, 0)
+	if len(zoneDomainMatchLabels) > 0 {
+		pts.Spec.TopologySpreadConstraints = append(pts.Spec.TopologySpreadConstraints, corev1.TopologySpreadConstraint{
```

Comment:
Should we add a similar constraint using topology key `kubernetes.io/hostname`? It'll find the most ideal node I believe as per this statement in the k8s doc:

"When a Pod defines more than one topologySpreadConstraint, those constraints are combined using a logical AND operation: the kube-scheduler looks for a node for the incoming Pod that satisfies all the configured constraints."

Reply:
Yes, I think it's okay to add here. It's been documented to not always work the way we expect (because the scheduler adds one pod at a time). But in such cases, we are covered by the `ScheduleAnyway` flag and our pod anti-affinity rules.